### ISSUE TITLE
Inconsistent use of `$target` in `popup.js`.

### ISSUE THREAD

**Issue by Parakleta** [Semantic-Org/Semantic-UI#4262]
I raised this issue previously in #3093 which was closed without being resolved.  I have since realised since commit 900f082387ba0614060008cbc2459c161312f27d fixed a typo in `popup.js` which broke my workaround that maybe my workaround was incorrect.

I find it quite confusing trying to work out what the expected interactions between the settings `.inline`, `.movePopup`, and `.popup` are supposed to be.

`.inline` can be set manually to specify that the popup should be created as a sibling of the *module*, or else is automatically set if the popup is found as a sibling of the *target*.  If an inline popup is found as a sibling of the *target*, then `.popup` is set to point to this popup.

`.movePopup` is always set, and when `.popup` is set it tries to move a popup to be a sibling of its *module*.  As discussed in #3093 this is the wrong behaviour when the popup was found to be a sibling of its *target* (i.e. automatic setting of `.inline`).

`.popup` seems to indicate that the popup originated in the document, or was dynamically created in a way that should be preserved.  Specifically note that this can be set manually, or else it is automatically set if the popup is found as a sibling of the *target*.  This also has the effect of disabling the `.removePopup()` method internally.

My previous solution had been to not add the inline popup to `.popup` and set `.preserve` instead, since `.preserve` and `.popup` seem to achieve similar outcomes in the sense that they prevent the popup being removed, but without the movement behaviour caused by `.popup` being set in `.refresh()`.  I think this is a mistake now, I think the issue is actually that the code seems to interchange in some places the handling of the *module* and the *target*.

The particularly interesting case is in `.refresh()` where if an inline popup is a sibling of the *target* then it is moved to be a sibling of the *module* and offset from the *module*, but if it is not a sibling of the *target* then it is not moved but is offset from the *target*.  Also interesting is in `.get.calculations()` where if `.popup` and `.movePopup` are set then it takes the same position data as if `.inline` is set, specifically the position of the *target* relative to its parent, rather than its offset relative to the document.

This suggests to me that the error is in [popup.js#L126](https://github.com/Semantic-Org/Semantic-UI/blob/961b4c90680e8d3c1e423fbf9c0fdaf20d1e87af/src/definitions/modules/popup.js#L126) where the `$offsetParent` value should be calculated from `$target` not from `$module`.  As an adjunct to this, [popup.js#L261](https://github.com/Semantic-Org/Semantic-UI/blob/961b4c90680e8d3c1e423fbf9c0fdaf20d1e87af/src/definitions/modules/popup.js#L261) should be inserting the popup as a sibling of `$target` and not `$module`.

I think if there is some requirement that the popup be left attached to the *module* (or elsewhere) and not rebound to the *target* then maybe this is the responsibility of the `.movePopup` setting, and otherwise it should always be assumed that if the popup exists (i.e. not dynamically created) then it must be a sibling of the *target*.

**Comment by jlukic** [comment]
As I said before, if there was a specific test case that showed how popup appeared incorrectly due to a miscalculation caused by mistaking `$target` and `$module` i'd be happy to evaluate it that way.

**Comment by Parakleta** [comment]
You removed the 'Needs Test Case' label and added the 'Confirmed Bug' label to the previous issue (#3093) when I provided a fiddle demonstrating the issue, and the bug demonstrated in that fiddle has never been fixed.  At this stage I don't know what else you want.  It is trivially provable both by example and by audit that the code does the wrong thing with regards to the `$target` and `$module` and I have provided a test case which shows it.  You yourself stated:

> It's clear to me that popup should be positioned next to settings.target (because the target might have a different coordinate system than activator), so that is indeed a bug.

Maybe the issue is that you are misunderstanding what it is that I am complaining about.  The problem is not the offset calculations which work fine.  The problem is moving the element through the DOM which messes up the CSS selectors, and also messes up things like React.  The popup element must be positioned as a sibling of the *target* and the inconsistent handling of '$module' and '$target' means that in some code paths it is moved to be attached as a sibling of the *module* instead of the *target*.

**Comment by jlukic** [comment]
Regardless of the veracity issue your tone is hostile, and not conducive to the free support you are receiving.

**Comment by Parakleta** [comment]
I'm not hostile, I'm frustrated.  You have an outstanding bug which you have agreed exists and yet you closed without resolving.  I have provided test cases, proposed solutions, I'm running out of ideas of what else to offer to get this resolved.  I have spent a lot of hours going through the source of this module trying to understand the code and trying to give you as much detail as I can, which I think is exactly conducive to the ethos of open source.

As it stands I'm not receiving free support, I'm offering it by producing for you tests cases and detailed bug reports.  I already have a forked version that I run because I needed this problem solved and I could have just used my version and walked away but I am trying to contribute back to this project for the benefit of the community.

I am really struggling to understand what happened between the point where #3093 became a confirmed bug and you decided to close it.
